activerecord/lib/active_record/enum.rb

```diff
@@ -182,7 +182,13 @@ def enum(definitions)
         detect_enum_conflict!(name, "#{name}=")
 
         attr = attribute_alias?(name) ? attribute_alias(name) : name
+        type, options = attributes_to_define_after_schema_loads[attr]
+
         attribute(attr, **default) do |subtype|
+          if type && !type.is_a?(Proc)
+            subtype = _lookup_cast_type(attr, type, options)
+          end
```

Comment:
This is similar to https://github.com/rails/rails/blob/71c8a89edd663dd21231f333ddb09b41bf94be37/activerecord/lib/active_record/attribute_methods/serialization.rb#L73-L78